common/src/parse.cpp

```diff
@@ -139,13 +149,12 @@ struct legally_representable {
     constexpr static bool value = legally_representable_v<T, U>;
 };
 
-using input_types = boost::mpl::vector<long int, long long int>;
-
-template <typename Integral>
-using legal_input_type_iter = typename boost::mpl::find_if<input_types,
-                                                           legally_representable<boost::mpl::_1, Integral>>::type;
+// assumptions:
+// * either long int or long long int is a valid type for storing Integral
+// * unsigned long long int is handled specially
 template <typename Integral>
-using legal_input_type = typename boost::mpl::deref<legal_input_type_iter<Integral>>::type;
+using legal_input_type = std::conditional_t<legally_representable_v<long int, Integral>,
+                                            long int, long long int>;
 
 static_assert ((std::is_same<legal_input_type<int>, long int>::value) ||
```

Comment:
This is a c++17 feature :/

Reply:
Atleast on MSVC according to [this](https://docs.microsoft.com/en-us/cpp/cpp/static-assert?view=vs-2019)

Reply:
Oh damn. I was following [standard](https://en.cppreference.com/w/cpp/types/conditional)

Reply:
I changed it to:
```
static_assert ((std::is_same<legal_input_type<int>, long int>::value) ||
               (std::is_same<legal_input_type<int>, long long int>::value),"int is not long int or long long int");
static_assert ((std::is_same<legal_input_type<unsigned int>, long int>::value) ||
               (std::is_same<legal_input_type<unsigned int>, long long int>::value),"Unsigned int is not long int or long long int");
```

Would it work at yours as well?

Reply:
Yes

Reply:
Actually the test doesn't make sense anymore. Since the conditional ensures that it will be either one of these. If the CI succeeds on this though, I'll send an update removing the test and a few other cosmetic changes